frontend/src/app/modules/work_packages/routing/partitioned-query-space-page/partitioned-query-space-page.component.ts

```diff
@@ -246,10 +246,7 @@ export class PartitionedQuerySpacePageComponent extends WorkPackagesViewBase imp
     }
 
     if (visibly) {
-      return this.loadingIndicator = promise.then((loadedQuery:QueryResource) => {
-        this.wpStatesInitialization.initialize(loadedQuery, loadedQuery.results);
```

Comment:
Interestingly the line was introduced only three months ago in PR https://github.com/opf/openproject/pull/8521. I don't know why that was necessary. So we need to look at it also from the perspective of that PR.

Reply:
Ok, it looks like the double loading must come from somewhere else.

Reply:
This part of the code feels wrong front to back. `additionalLoadingTime` doesn't add any extra loading time, afaik. It's a promise that instantly resolves. However, this puts the callback into the microtask queue, so I didn't want to leave the behaviour out out of fear of breaking unrelated stuff. 

The query should be getting loaded from `wp-list.service.ts:84`, where a new `queryRequest` triggers a state initialization. Before my change, this resulted (via a 5-service round-trip) into a second state initialization in the partitioned query space component.

So it seems there are two different entrypoints into the query-loading. If I remove the `initialize` call from `wp-list.service`, the calendar widget won't load, among other stuff. It seems like this double-loading issue warrants a rewrite that is too large for the next release.

Reply:
`additionalLoadingTime` was added when the timeline was added to ensure the loading indicator remains until the timeline itself is rendered (blocking users from editing in between). It is used in this subclass https://github.com/opf/openproject/blob/6d82c32fcc07aaf8dba7c018a7c2dd881b705a9d/frontend/src/app/modules/work_packages/routing/wp-view-page/wp-view-page.component.ts#L101

This whole thing is a mess that Aleix and I have talked quite a bit how to refactor it, but this will require a huge effort. I suggest to let us three discuss this when I'm back in January.

Reply:
Thanks @oliverguenther